### PR TITLE
Move Fonts Library to compat dir for 6.5

### PR DESCRIPTION
**Pull request by getdave**
## What?

As the Fonts Library is slated for WP 6.5 this PR moves the files into the `compat/wordpress-6.5` dir.

## Why?

It's then easier to see which files can be removed once the minimum WP Core version in Gutenberg gets updated.

## How?

Moves all files.

### CODE REVIEW

**Comment by getdave** [comment]
I'm unsure as to what should happen with the "bc" layer code. Can someone validate this please 🙏 ?

**Comment by hellofromtonya** [comment]
Thanks @getdave for creating this PR.  A few change suggestions:

* Leave the `font-face/bc-layer/` in the experimental folder, as it's for the plugin only.
* Move the `font-library.php` file to its parent `fonts` and rename it `fonts.php`:  `fonts/font-library/fonts.php`. Why? The new global functions in that file will go into [Core's `wp-includes/fonts.php` file](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/fonts.php).

In the Core PR, the files will go:

- The new functions -> `wp-includes/fonts.php` ([see here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/fonts.php))

- The class files -> `wp-includes/fonts/` without adding a `font-library` subdirectory ([see here](https://github.com/WordPress/wordpress-develop/tree/trunk/src/wp-includes/fonts))

**Comment by youknowriad** [comment]
Makes sense @hellofromtonya I applied the feedback and I also removed the useless font-library subfolder.

**Review by youknowriad — APPROVED** [review]

**Review by anton-vlasenko — APPROVED** [review]
The code looks good to me.
I'm fine with getting rid of the `lib/experimental/fonts/font-face/bc-layer/` folder and moving it to `lib/experimental/font-face/bc-layer/`, as long as it doesn't bear any hidden risks—and this change looks safe.

**Comment by getdave** [comment]
💥

**Comment by getdave** [comment]
Let's rebase all the other "nit' PRs.

File: lib/load.php [hunk 1]
```diff
@@ -160,18 +160,18 @@ function gutenberg_is_experiment_enabled( $name ) {
...
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-fonts.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-fonts-provider-local.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-fonts-resolver.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-gutenberg-fonts-api-bc-layer.php';
+require __DIR__ . '/experimental/font-face/bc-layer/webfonts-deprecations.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-webfonts-utils.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-webfonts-provider.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-webfonts-provider-local.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-webfonts.php';
+require __DIR__ . '/experimental/font-face/bc-layer/class-wp-web-fonts.php';
```
Comment (Mamaduka) [inline lib/load.php:174]: Do we have a timeline for removing this BC layer?

**Pull request merged**